### PR TITLE
Add i18n to emoji picker

### PR DESCRIPTION
**Pull request by MartinCupela**
Adding optional `emojiI18n` prop to `StatusUpdateFormProps` in order the user can pass custom strings / translations to EmojiPicker.

### CODE REVIEW

**Review by mahboubii — COMMENTED** [review]
LGTM, Could you also add some tests for this?

**Review by arnautov-anton — APPROVED** [review]

**Comment by MartinCupela** [comment]
> LGTM, Could you also add some tests for this?

@mahboubii will do :)

**Comment by MartinCupela** [comment]
@arnautov-anton @mahboubii Finally added those tests to EmojiPicker + StatusUpdateForm. I have only tested that the props are correctly passed down. Could you please review these changes? Thank you

File: src/mock-builders/TranslationContext.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React, { ReactNode } from 'react';
```
Comment (arnautov-anton) [inline src/mock-builders/TranslationContext.tsx:1]:
```suggestion
import React, { PropsWithChildren } from 'react';
```

File: src/mock-builders/TranslationContext.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import moment from 'moment';
+import { TranslationContextValue, TranslationProvider } from '../context';
+
+export const translationProviderData = { t: String, tDateTimeParser: moment };
+
+type TranslationProviderMockProps = { children?: ReactNode; value?: TranslationContextValue };
+
+export const TranslationProviderMock = ({ value, children }: TranslationProviderMockProps) => (
+  <TranslationProvider value={value || translationProviderData}>{children}</TranslationProvider>
+);
```
Comment (arnautov-anton) [inline src/mock-builders/TranslationContext.tsx:11]:
```suggestion
export const TranslationProviderMock = ({ value = translationProviderData, children }: TranslationProviderMockProps) => (
  <TranslationProvider value={value}>{children}</TranslationProvider>
);
```

**Comment by arnautov-anton** [comment]
LGTM. 👍

**Review by arnautov-anton — DISMISSED** [review]

File: src/i18n/Streami18n.ts [hunk 3]
```diff
@@ -179,6 +179,11 @@ export type TDateTimeParser = (input?: string | number | Date) => Dayjs.Dayjs |
 
 export type LanguageCallbackFn = (t: TFunction) => void;
 
+export type Translator = {
+  t: TFunction;
+  tDateTimeParser: TDateTimeParser;
+};
+
```
Comment (mahboubii) [inline src/i18n/Streami18n.ts:186]: could be refactored a bit since a similar type already exists `TranslationContextValue`

File: package.json [hunk 4]
```diff
@@ -41,7 +41,8 @@
     "prepare": "yarn run build && husky install",
     "preversion": "yarn install && yarn lint && yarn test",
     "version": "yarn docs-build && git add -A docs",
-    "postversion": "git push && git push --tags && npm publish"
+    "postversion": "git push && git push --tags && npm publish",
+    "watch": "watch 'yarn build' ./src/components --wait=3"
```
Comment (mahboubii) [inline package.json:45]: watch dep needs to be added to devDependencies 
```suggestion
    "watch": "watch 'yarn build' ./src --wait=3"
```

File: src/components/EmojiPicker.tsx [hunk 5]
```diff
@@ -1,23 +1,48 @@
...
   emojiData?: EmojiDataSet;
+  i18n?: Partial<I18n>;
   onSelect?: (emoji: EmojiData) => void;
 };
 
-export const EmojiPicker = ({ emojiData = defaultEmojiData, onSelect }: EmojiPickerProps) => {
+export const getEmojiPickerFieldsTranslations = (t: TFunction): I18n => ({
+  search: t('Search'),
+  // todo: remove @ts-ignore after fixed I18n type definition in emoji-mart package
+  // @ts-ignore
```
Comment (mahboubii) [inline src/components/EmojiPicker.tsx:25]:
```suggestion
  // todo: remove after fixed I18n type definition in emoji-mart package
  // @ts-expect-error
```

File: package.json (hunk 4 above)
Comment (mahboubii) [reply package.json:45]: actually rollup already has a watch mode that can be used to rebuild the lib

File: src/components/StatusUpdateForm/StatusUpdateForm.test.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    const emojiI18n: Partial<I18n> = {
+      search: 'Custom Search String',
+      // @ts-ignore
+      categories: { recent: 'Recent Emojis' },
+    };
+
+    const context = {};
+
+    render(<StatusUpdateForm emojiI18n={emojiI18n} />);
+    expect(EmojiPicker).toHaveBeenCalledWith(expect.objectContaining({ i18n: emojiI18n }), context);
```
Comment (mahboubii) [inline src/components/StatusUpdateForm/StatusUpdateForm.test.tsx:25]:
```suggestion
    render(<StatusUpdateForm emojiI18n={emojiI18n} />);
    expect(EmojiPicker).toHaveBeenCalledWith(expect.objectContaining({ i18n: emojiI18n }), {});
```

File: src/mock-builders/TranslationContext.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React, { PropsWithChildren } from 'react';
+import moment from 'moment';
+import { TranslationContextValue, TranslationProvider } from '../context';
+
+export const translationProviderData = { t: String, tDateTimeParser: moment };
+
+type TranslationProviderMockProps = PropsWithChildren<{ value?: TranslationContextValue }>;
+
+export const TranslationProviderMock = ({
```
Comment (mahboubii) [inline src/mock-builders/TranslationContext.tsx:9]: mocking this is nice but it's also possible to just use the `TranslationProvider` with a default value directly in the test

File: src/i18n/en.json [hunk 8]
```diff
@@ -19,12 +19,11 @@
   "No data to display...": "No data to display...",
   "No emoji found": "No emoji found",
   "Objects": "Objects",
-  "People & Body": "People & Body",
   "Pick your emoji": "Pick your emoji",
   "Post": "Post",
   "Search": "Search",
   "Search Results": "Search Results",
-  "Smileys & Emotion": "Smileys & Emotion",
+  "Smileys & People": "Smileys & People",
```
Comment (mahboubii) [inline src/i18n/en.json:26]: is this change necessary to adapt to emoji-mart default set?

File: package.json (hunk 4 above)
Comment (MartinCupela) [reply package.json:45]: Yes, this is my mistake. I actually didn't want to commit this one. I will remove this script

File: src/components/StatusUpdateForm/StatusUpdateForm.test.tsx (hunk 6 above)
Comment (MartinCupela) [reply src/components/StatusUpdateForm/StatusUpdateForm.test.tsx]: Removin context

File: src/mock-builders/TranslationContext.tsx (hunk 7 above)
Comment (MartinCupela) [reply src/mock-builders/TranslationContext.tsx:9]: Removing the mock altogether.

File: src/i18n/Streami18n.ts (hunk 3 above)
Comment (MartinCupela) [reply src/i18n/Streami18n.ts:186]: Removing the duplicate. Thank you

File: src/i18n/en.json (hunk 8 above)
Comment (MartinCupela) [reply src/i18n/en.json:26]: @mahboubii the intention was not only to align it with the npm package, but also with the key under which it is stored in the I18n object (`categories.people`). Do you prefer me to revert this change?

Comment (MartinCupela) [reply src/i18n/en.json]: reverting back to Smileys & Emotion

**Review by arnautov-anton — APPROVED** [review]
